internal/plugins/testing/agent_mock.go

```diff
@@ -18,6 +18,7 @@ import (
 )
 
 type MockAgent struct {
+	agent.AgentContext
```

Comment:
why?

Reply:
there's no embedded implementation of the interface and therefore it'd be safer to have the compiler complain about lacking implementations at compile time I think